roseau/load_flow_single/models/core.py

```diff
@@ -167,14 +167,30 @@ def _res_getter(self, value: _T | None, warning: bool) -> _T:
         self._fetch_results = False
         return value
 
+    def _check_geometry(self, geometry: shapely.Geometry | None) -> BaseGeometry | None:
+        """Check if the geometry is a valid shapely geometry."""
+        # shapely.Geometry is the public name of shapely objects but it has no attributes. We accept
+        # shapely.Geometry as input because users most likely use it as annotation but internally we
+        # use BaseGeometry because it has all the shapely geometry attributes.
+        # In practice, there are no shapely.Geometry objects that are not BaseGeometry objects.
+        if geometry is None:
+            return None
+        if not isinstance(geometry, BaseGeometry):
+            msg = (
+                f"The geometry of {self._element_info} is not a valid shapely geometry. Got {type(geometry).__name__}."
+            )
+            logger.error(msg)
+            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_GEOMETRY_TYPE)
+        return geometry
```

Comment:
I'm not convinced of this code, if we accept only `BaseGeometry` and not `Geometry` only, shouldn't we use the first to make things clear ? 
I've searched on github and most people seem to use `BaseGeometry` anyway:
`/(?-i)from shapely import.* Geometry[,\n]/ ` -> 138 results
`/(?-i)from shapely.geometry.base import BaseGeometry[,\n]/` -> 2.3k results

Reply:
It seems that Base Geometry is a subclass of Geometry. See [here](https://github.com/shapely/shapely/blob/6166fd54bfc72d5be2394dfc03bbcddb6016a62a/shapely/geometry/base.py#L97). As far as I remember, it has not always been the case...

Anyway, we could be able to use `isinstance(geometry, Geometry)` everywhere. `Geometry` has the advantage to be exposed in the shapely module directly.

Reply:
I changed the comment and restored the `BaseGeometry` annotations.